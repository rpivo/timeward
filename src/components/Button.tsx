import React from 'react';
import StyledButton from '@styles/components/Button.styled';
import buttonPaths from '@utilities/svg';

export type ButtonProps = {
  kind: string;
  handleClick: () => void;
}

const Button: React.FC<ButtonProps> = ({ kind, handleClick }: ButtonProps): JSX.Element => {

  const Start: React.FC = (): JSX.Element =>
    <svg width="15" height="15" viewBox="0 0 50 50">
      <path d={ buttonPaths.start } transform="translate(54.392 -7.309) rotate(90)" fill="#aaa" />
    </svg>;

  const Pause: React.FC = (): JSX.Element =>
    <svg width="15" height="15" viewBox="0 0 50 50">
      <g transform="translate(33 404)">
        <path d={ buttonPaths.pause } transform="translate(-5.034 -404)" fill="#aaa"/>
        <path d={ buttonPaths.pause } transform="translate(-33 -404)" fill="#aaa"/>
      </g>
    </svg>;

  const Stop: React.FC = (): JSX.Element =>
    <svg width="15" height="15" viewBox="0 0 50 50">
      <path d={ buttonPaths.stop } fill="#aaa"/>
    </svg>;

  const renderSwitch = (buttonType: string): JSX.Element => {
    switch (buttonType) {
    case 'start':
      return <Start />;
    case 'pause':
      return <Pause />;
    default:
      return <Stop />;
    }
  };

  return (
    <StyledButton kind={ kind } onClick={ handleClick }>
      { renderSwitch(kind) }
    </StyledButton>
  );
};

export default Button;
