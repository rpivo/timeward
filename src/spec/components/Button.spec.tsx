import React from 'react';
import renderer from 'react-test-renderer';
import Button from '@components/Button';

describe('Button', () => {

  afterEach(() => {
    jest.restoreAllMocks();
  });

  const button = <Button kind='pause' handleClick={ (): null => null }/>;

  describe('render', () => {
    const testRenderer = renderer.create(button);

    it('should render correctly', () => {
      testRenderer.toJSON();
      expect(testRenderer).toMatchSnapshot();
    });

  });
});
