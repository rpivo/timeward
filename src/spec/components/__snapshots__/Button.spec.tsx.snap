// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render should render correctly 1`] = `
<div
  className="sc-AykKC kZlTAg"
  onClick={[Function]}
>
  <svg
    height="50"
    viewBox="0 0 41.949 50"
    width="41.949"
  >
    <g
      transform="translate(33 404)"
    >
      <path
        d="M4.237,0H9.746a4.237,4.237,0,0,1,4.237,4.237V45.763A4.237,4.237,0,0,1,9.746,50H4.237A4.237,4.237,0,0,1,0,45.763V4.237A4.237,4.237,0,0,1,4.237,0Z"
        fill="#aaa"
        transform="translate(-5.034 -404)"
      />
      <path
        d="M4.237,0H9.746a4.237,4.237,0,0,1,4.237,4.237V45.763A4.237,4.237,0,0,1,9.746,50H4.237A4.237,4.237,0,0,1,0,45.763V4.237A4.237,4.237,0,0,1,4.237,0Z"
        fill="#aaa"
        transform="translate(-33 -404)"
      />
    </g>
  </svg>
</div>
`;
